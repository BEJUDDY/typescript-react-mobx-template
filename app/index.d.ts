/// <reference path="node_modules/rx/ts/rx.all.d.ts" />
/// <reference types="jest" />
declare const process: any;