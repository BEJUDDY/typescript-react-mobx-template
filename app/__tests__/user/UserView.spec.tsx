import * as React from 'react';
import {create} from 'react-test-renderer';
import {diContext} from '../../src/scripts/app/DIContext';
import {UsersView} from '../../src/scripts/user/UsersView';
import * as TypeMoq from 'typemoq';
import {UserStore, UserViewStore} from '../../src/scripts/user/UserStore';
import {User} from '../../src/scripts/user/model';
import {BrowserRouter as Router} from 'react-router-dom';

it('UsersView should render proper DOM', () => {
    const user1: User = {
        id: 1,
        firstName: "test",
        lastName: "test"
    };
    const user2: User = {
        id: 2,
        firstName: "test2",
        lastName: "test2"
    };

    const userStoreMock = TypeMoq.Mock.ofType<UserStore>();
    const userViewStoreMock = TypeMoq.Mock.ofType<UserViewStore>();
    userStoreMock.setup(x => x.createUserViewStore()).returns(() => userViewStoreMock.object);
    userViewStoreMock.setup(x => x.users).returns(() => [user1, user2]);

    diContext.userStore = userStoreMock.object;

    expect(create(<Router><UsersView/></Router>).toJSON()).toMatchSnapshot();
});