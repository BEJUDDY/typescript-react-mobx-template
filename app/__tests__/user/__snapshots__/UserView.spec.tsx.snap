exports[`test UsersView should render proper DOM 1`] = `
<div>
  <table
    className="table">
    <thead>
      <tr>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          test
        </td>
        <td>
          test
        </td>
        <td>
          <a
            className="btn btn-primary"
            href="/user/1"
            onClick={[Function]}>
            <span
              aria-hidden={true}
              className="fa fa-info-circle" />
          </a>
        </td>
      </tr>
      <tr>
        <td>
          test2
        </td>
        <td>
          test2
        </td>
        <td>
          <a
            className="btn btn-primary"
            href="/user/2"
            onClick={[Function]}>
            <span
              aria-hidden={true}
              className="fa fa-info-circle" />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
